realm/realm-library/src/androidTest/kotlin/io/realm/RealmModelManagedSetTester.kt

```diff
@@ -0,0 +1,394 @@
+            assertFailsWith<NullPointerException> {
+                set.addAll(TestHelper.getNull())
+            }
+        }
+
+
+        // Test with objects from another realm
+        objectsFromOtherThread {
+            set.addAll(it)
+        }
+    }
+
+    override fun retainAll() {
+        // Test with managed realm objects
+        managedTester.retainAll()
+
+        // Test with unmanaged realm objects
+        val set = initAndAssertEmptySet()
+        realm.executeTransaction {
+            // Check throws exception when unmanaged values are passed
+            assertFailsWith<IllegalArgumentException>("Collection with unmanaged objects not permitted") {
+                set.retainAll(unmanagedInitializedSet)
+            }
+
+            // Check throws exception when null values are passed
+            assertFailsWith<java.lang.NullPointerException>("Collections with nulls are not permitted") {
+                set.retainAll(listOf(null))
+            }
+        }
+
+
+        // Test with objects from another realm
+        objectsFromOtherThread {
+            set.retainAll(it)
+        }
+    }
+
+    override fun removeAll() {
+        // Test with managed realm objects
+        managedTester.removeAll()
+
+        // Test with unmanaged realm objects
+        val set = initAndAssertEmptySet()
+        realm.executeTransaction {
+            // Check throws exception when unmanaged values are passed
+            assertFailsWith<IllegalArgumentException>("Collection with unmanaged objects not permitted") {
+                set.removeAll(unmanagedInitializedSet)
+            }
+
+            // Check throws exception when null values are passed
+            assertFailsWith<java.lang.NullPointerException>("Collections with nulls are not permitted") {
+                set.removeAll(listOf(null))
+            }
+        }
+
+        // Test with objects from another realm
+        objectsFromOtherThread {
+            set.removeAll(it)
+        }
+    }
+
+    override fun clear() = managedTester.clear()
+
+    override fun freeze() = managedTester.freeze()
+
+    override fun setters() {
+        managedTester.setters()
+
+        // Test with objects from another realm
+        looperThread.runBlocking {
+            Realm.getInstance(realm.configuration).use { alternativeRealm ->
+                val alternativeObject = createAllTypesManagedContainerAndAssert(alternativeRealm, "alternativeObject")
+
+                alternativeRealm.executeTransaction {
+                    val alternativeSet = RealmSet<T>().init(managedInitializedSet)
+
+                    assertFailsWith<IllegalArgumentException>("Cannot pass values from another Realm") {
+                        setSetter(alternativeObject, alternativeSet)
+                    }
+                }
+            }
+            looperThread.testComplete()
+        }
+    }
+
+    private fun objectsFromOtherThread(block: (values: List<T?>) -> Unit) {
```

Comment:
Maybe rename so that it is obvious that you are actually doing the tests and playing around with the looper. It is not clear at the call site that it is doing a heavy lift.